src/core/function.jl

```diff
@@ -10,6 +10,8 @@ register(Function, API.LLVMFunctionValueKind)
 Function(mod::Module, name::String, ft::FunctionType) =
     Function(API.LLVMAddFunction(mod, name, ft))
 
+FunctionType(Fn::Function) = FunctionType(API.LLVMGetFunctionType(Fn))
```

Comment:
We can do this already using `eltype(llvmtype(fn))`. Probably wouldn't hurt to have the `FunctionType` ctor for visibility, but it doesn't need an additional API call.

Reply:
Oh interesting; that will require a lot of updates across the stack then (we're doing `eltype(llvmtype(...))` in a lot of places).

Reply:
`llvmeltype` is probably illegal everywhere.

Reply:
Still valid for Arrays, Vectors, etc.

Reply:
In any case illegal uses should be caught by that error method. Otherwise it segfaults

Reply:
`function_type` sounds good. Maybe we should rename `llvmtype` to `value_type` then? Or that may be too breaking for what it's worth.

Reply:
llvmtype isn't too bad, and `value_type` isn't super clear IMO, and `llvm_value_type` is a handful.